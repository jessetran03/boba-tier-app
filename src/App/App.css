* {
  box-sizing: border-box;
}

body {
  font-family: sans-serif;
  font-size: 20px;
  color: #444444;
  text-align: center;
  min-width: 300px;
}

a:link {
  color: #555555;
  text-decoration: none;
}

a:visited {
  color: #555555;
  text-decoration: none;
}

a:hover {
  color: #000000;
  text-decoration: underline;
}

a:active {
  color: #EEEEEE;
  text-decoration: underline;
}