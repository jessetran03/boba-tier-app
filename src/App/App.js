import React, { Component } from 'react';
import { Route } from 'react-router-dom'
import Landing from '../pages/Landing/Landing'
import Login from '../pages/Login/Login'
import Register from '../pages/Register/Register'
import Rankings from '../pages/Rankings/Rankings'
import Shop from '../pages/Shop/Shop'
import Nav from '../components/Nav/Nav'
import './App.css'

/*{
  ['/workouts/:workoutId'].map(path =>
    <Route
      exact
      key={path}
      path={path}
      component={WorkoutExerciseList}
    />
  )
}*/

class App extends Component {
  renderMainRoutes() {
    return (
      <>
        <Route
          exact
          path='/'
          component={Landing}
        />
        <Route
          exact
          path='/login'
          component={Login}
        />
        <Route
          exact
          path='/register'
          component={Register}
        />
        <Route
          path='/rankings'
          component={Rankings}
        />
        {['/shop/:shopId'].map(path =>
          <Route
            exact
            key={path}
            path={path}
            component={Shop}
          />
        )}
      </>
    )
  }

  render() {
    return (
      <>
        <Route
          path='/'
          component={Nav}
        />
        <main>
          {this.renderMainRoutes()}
        </main>
      </>
    )
  }
}

export default App;