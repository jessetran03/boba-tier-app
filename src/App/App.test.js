import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import App from './App'
import renderer from 'react-test-renderer'

it('renders App without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render App without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  )
    .toJSON();
    expect(wrapper).toMatchSnapshot()
  })