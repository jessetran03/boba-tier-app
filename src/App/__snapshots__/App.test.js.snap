// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App without data 1`] = `
Array [
  <nav>
    <section
      className="nav-header"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h2>
          Boba Tier
        </h2>
      </a>
    </section>
    <section
      className="nav-links"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <p>
          Home
        </p>
      </a>
      <a
        aria-current={null}
        href="/shops"
        onClick={[Function]}
      >
        <p>
          Shop List
        </p>
      </a>
      <a
        aria-current={null}
        href="/rankings"
        onClick={[Function]}
      >
        <p>
          Rankings
        </p>
      </a>
      <a
        aria-current={null}
        href="/shops/2"
        onClick={[Function]}
      >
        <p>
          Sharetea
        </p>
      </a>
      <select>
        <option
          value="Houston"
        >
          Houston
        </option>
      </select>
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
      >
        <p>
          Log In
        </p>
      </a>
      <a
        aria-current={null}
        href="/register"
        onClick={[Function]}
      >
        <p>
          Sign Up
        </p>
      </a>
    </section>
  </nav>,
  <main>
    <section>
      <section
        className="landing"
      >
        Rate the best boba tea in town. Pick a city to get started.
      </section>
      <ul
        className="landing-cities"
      >
        <li>
          <a
            href="/shops"
            onClick={[Function]}
          >
            <button>
              Houston
            </button>
          </a>
        </li>
      </ul>
      <p>
        More cities to come later!
      </p>
    </section>
  </main>,
]
`;
